modules/gdscript/gdscript_byte_codegen.cpp

```diff
@@ -143,6 +143,14 @@ void GDScriptByteCodeGenerator::pop_temporary() {
 	ERR_FAIL_COND(used_temporaries.is_empty());
 	int slot_idx = used_temporaries.back()->get();
 	const StackSlot &slot = temporaries[slot_idx];
+	if (slot.type == Variant::OBJECT) {
+		// Avoid keeping in the stack long-lived references to objects,
+		// which may prevent RefCounted objects from being freed.
+		GDScriptDataType type = {};
+		type.kind = GDScriptDataType::BUILTIN;
+		type.builtin_type = slot.type;
```

Comment:
Setting the type here is not needed. This won't be used by `write_assign_false()`.

Reply:
So can I just do `write_assign_false(Address(Address::TEMPORARY, slot_idx));`?

Reply:
Yes.